helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -705,49 +706,80 @@ public void setInstanceCapacityKeys(List<String> capacityKeys) {
 
   /**
    * Get the default instance capacity information from the map fields.
+   *
    * @return data map if it exists, or empty map
    */
   public Map<String, Integer> getDefaultInstanceCapacityMap() {
-    Map<String, String> capacityData =
-        _record.getMapField(ClusterConfigProperty.DEFAULT_INSTANCE_CAPACITY_MAP.name());
-
-    if (capacityData != null) {
-      return capacityData.entrySet().stream().collect(
-          Collectors.toMap(entry -> entry.getKey(), entry -> Integer.parseInt(entry.getValue())));
-    }
-    return Collections.emptyMap();
+    return getDefaultCapacityMap(ClusterConfigProperty.DEFAULT_INSTANCE_CAPACITY_MAP);
   }
 
   /**
    * Set the default instance capacity information with an Integer mapping.
+   *
    * @param capacityDataMap - map of instance capacity data
    * @throws IllegalArgumentException - when any of the data value is a negative number or when the map is empty
-   *
-   * This information is required by the global rebalancer.
+   *                                  This information is required by the global rebalancer.
    * @see <a href="Rebalance Algorithm">
-   *   https://github.com/apache/helix/wiki/Design-Proposal---Weight-Aware-Globally-Even-Distribute-Rebalancer#rebalance-algorithm-adapter
-   *   </a>
+   * https://github.com/apache/helix/wiki/Design-Proposal---Weight-Aware-Globally-Even-Distribute-Rebalancer#rebalance-algorithm-adapter
+   * </a>
    * If the instance capacity is not configured in neither Instance Config nor Cluster Config, the
    * cluster topology is considered invalid. So the rebalancer may stop working.
```

Comment:
Is this description valid anymore? We aren't throwing an exception anymore.

Reply:
We throw an exception while the set map parameters are invalid.
As for the rebalancer, we are not throwing an exception. But the new assignment won't be calculated (rebalance stopped) on invalid cluster status. So the description is still valid.